Convert to circleci v2 config

h-lame opened the pull request:
We started by using the [config-translation endpoint][config-translation] that
circleci provides to convert our existing v1 config into the new v2 config.  It
contained lots of extra stuff and comments so I removed a load of them and
simplified it.

Note that we need to run our specs on a ruby image that matches the version
specified in our `.ruby-version`, and that we also need a ruby image that has
a JS runtime, so we use `circleci/ruby:2.3.7-node`.

[config-translation]: https://circleci.com/docs/2.0/config-translation/

h-lame committed:
Convert to circleci v2 config

We started by using the [config-translation endpoint][config-translation] that
circleci provides to convert our existing v1 config into the new v2 config.  It
contained lots of extra stuff and comments so I removed a load of them and
simplified it.

Note that we need to run our specs on a ruby image that matches the version
specified in our `.ruby-version`, and that we also need a ruby image that has
a JS runtime, so we use `circleci/ruby:2.3.7-node`.

[config-translation]: https://circleci.com/docs/2.0/config-translation/

ubxd commented:
Deploy preview for *unboxed-co-production* ready!

Built with commit 0e6b0d69e0672bf520ff6e722cc7912fead81080

https://deploy-preview-492--unboxed-co-production.netlify.com

h-lame committed:
Don't run in parallel while building site in CI

This will make it slower, but in the new circleci v2 world I was seeing a lot
of failures because one (or more) of the parallel workers used to build the site
were failing to load the middleman command.

h-lame committed:
Cache gems based on Gemfile.lock not branch name

If we cache the gems based on the Gemfile.lock instead of the branch name then
we should be able to avoid bundle install steps for branches that don't change
any gems.
